### PR TITLE
Future compatibility

### PR DESCRIPTION
**Pull request by mkolopanis**
Making sure the code can run in py2 and py3, but doing to so by hand so expect some trial/error.

Also making some of our scripts more human readable.

### CODE REVIEW

**Comment by adampbeardsley** [comment]
I did a quick test running:
`python LWA_bifrost.py --offline --tbnfile=/data5/LWA_SV_data/data_raw/TBN/Jupiter/058161_000086727 --imagesize 64 --imageres 1.79057 --nts 512 --channels 4 --accumulate 50 --ints_per_file 40`
on master and on this branch, and the output images are identical.
I also ran the same using `LWA_bifrost_DFT.py`, and also got identical images between master and this branch.

**Comment by adampbeardsley** [comment]
^ note, I ran these in py2. We'll need to integrate the LSL API changes before running in py3. There's a whole process.

**Comment by adampbeardsley** [comment]
@HaraKrish would you mind taking a look at this PR? Just get one more set of eyes before we merge in.

File: LWA/LWA_bifrost.py [hunk 1]
```diff
@@ -1322,16 +1573,19 @@ def main():
...
     if not os.path.isdir(args.out_dir):
-        print('Output directory does not exist. Defaulting to current directory.')
-        args.out_dir = '.'
+        print("Output directory does not exist. Defaulting to current directory.")
+        args.out_dir = "."
 
     if args.removeautocorrs:
-        raise NotImplementedError('Removing autocorrelations is not yet properly'
-                                  + ' implemented.')
+        raise NotImplementedError(
```
Comment (HaraKrish) [inline LWA/LWA_bifrost.py:1580]: I have a new module aCorr in epic_optim branch that replaces the bfMap call for autocorrelation. It works fine so the "removeautocorrs" flag can be enabled again.

Comment (adampbeardsley) [reply LWA/LWA_bifrost.py:1580]: Let's do that in a separate PR. We want to keep things that are functionally different in separate merges.

File: LWA/LWA_bifrost.py [hunk 2]
```diff
@@ -844,109 +942,149 @@ def main(self):
...
-                                        shape=(self.ntime_gulp, nchan, npol**2, self.grid_size, self.grid_size))
-
+                                bifrost.map(
+                                    "a(i,j,k,l) += (b(i,j,k,l/2) * b(i,j,k,l%2).conj())",
+                                    {"a": autocorrs, "b": udata, "t": self.ntime_gulp},
+                                    axis_names=("i", "j", "k", "l"),
+                                    shape=(self.ntime_gulp, nchan, nstand, npol ** 2),
+                                )
+
+                            bifrost.map(
```
Comment (HaraKrish) [inline LWA/LWA_bifrost.py:1044]: You can replace this with the following :

                           ## Cross-Correlation of Antennas
                            if self.benchmark == True:
                                timecorr1 = time.time()

                            try:
                                bf_xcorr.execute(gdata, crosspol)
                            except NameError:
                                bf_xcorr = xCorr()
                                bf_xcorr.init(self.locs,self.grid_size, polmajor=False)
                                bf_xcorr.execute(gdata, crosspol)

                            if self.benchmark == True:
                                timecorr2 = time.time()
                                print(" Cross-Corr time : %f" % (timecorr2 - timecorr1))

File: LWA/LWA_bifrost.py [hunk 3]
```diff
@@ -844,109 +942,149 @@ def main(self):
...
-                                            {'a':autocorrs, 'b':udata,'t':self.ntime_gulp},
-                                            axis_names=('i','j','k','l'),
-                                            shape=(self.ntime_gulp,nchan,nstand,npol**2))
-
-                            bifrost.map('a(i,j,p,k,l) += b(0,i,j,p/2,k,l)*b(0,i,j,p%2,k,l).conj()',
-                                        {'a':crosspol, 'b':gdata},
-                                        axis_names=('i','j', 'p', 'k', 'l'),
-                                        shape=(self.ntime_gulp, nchan, npol**2, self.grid_size, self.grid_size))
-
+                                bifrost.map(
```
Comment (HaraKrish) [inline LWA/LWA_bifrost.py:1037]: Like-wise you can replace this function call with following : 

                               ## Auto-Correlation of Antennas
                                if self.benchmark == True:
                                  timeacorr1 = time.time()

                                try:
                                  bf_acorr.execute(udata, autocorrs)
                                except NameError:
                                  bf_acorr = aCorr()
                                  bf_acorr.init(self.locs, polmajor=False)
                                  bf_acorr.execute(udata, autocorrs)

                                if self.benchmark == True:
                                  timeacorr2 = time.time()
                                  print(" Auto-Corr time : %f" % (timeacorr2 - timeacorr1))

File: LWA/LWA_bifrost.py [hunk 4]
```diff
@@ -1,65 +1,70 @@
...
 from bifrost.quantize import quantize as Quantize
-from bifrost.reduce import reduce as Reduce
 from bifrost.proclog import ProcLog
 from bifrost.libbifrost import bf
 from bifrost.fft import Fft
 from bifrost.romein import Romein
-import bifrost
-import bifrost.affinity
-from bifrost.libbifrost import bf
+
```
Comment (HaraKrish) [inline LWA/LWA_bifrost.py:41]: from bifrost.aCorr import aCorr
from bifrost.xCorr import xCorr

Comment (HaraKrish) [reply LWA/LWA_bifrost.py:41]: These modules are tested and part of epic_optim branch which can be used and incorporated.

File: LWA/LWA_bifrost.py (hunk 1 above)
Comment (HaraKrish) [reply LWA/LWA_bifrost.py:1580]: Ah ok, I already tested the new code in py2.. Is it possible to test in py3 given that we don't have a functional py3 bifrost yet

Comment (adampbeardsley) [reply LWA/LWA_bifrost.py:1580]: No, only py2 right now. But you're proposing changes to the code that aren't relevant to this PR. this is just changing syntax to be py3 compatible without changing any functionality. We should have a separate PR for your updates. That will make tracking the history easier.

Comment (HaraKrish) [reply LWA/LWA_bifrost.py:1580]: Ok, got it. I've removed those comments.

**Pull request merged**